nrf_security/src/CMakeLists.txt

```diff
@@ -184,6 +189,8 @@ append_with_prefix(src_crypto_base ${ARM_MBEDTLS_PATH}/library
   platform_util.c
   version.c
   constant_time.c
+
+  hkdf.c
```

Comment:
Shouldn't this only be added when MBEDTLS_HKDF_C is enabled?

I understand that we always need it when TF-M is enabled, but this will add hkdf.c even for non-TFM builds wont it?

Reply:
Will investigate this.

Reply:
Moved the addition of this source file to a more appropriate location, and added if guards for the special case needed.